contrib/nginx.md

```diff
@@ -0,0 +1,106 @@
+# Setting up nginx as a proxy to soju
+
+You can configure nginx as a proxy to soju to handle both public http/tls and also irc/tls connections.
+This way we don't need to expose soju listening sockets publicly and relieve it from handling tls.
+One benefit of this approach would be that soju doesn't need to be hooked into certbot, nor does it need
+to be reloaded when certificates are updated. Potentially the nginx TLS implementation might be more robust
+but that hasn't been proven :D.
+
+In this example, nginx will also handle serving static files for gamja, from the `/srv/gamja` directory.
+
+**Note:** different Linux distributions use a different configuration structure for nginx, so here we only
+provide the configuration snippets for the [`http`/`server`](https://nginx.org/en/docs/http/ngx_http_core_module.html#server)
+module, the [http proxy](https://nginx.org/en/docs/http/ngx_http_proxy_module.html) configuration and
+the [`stream`/`server`](https://nginx.org/en/docs/stream/ngx_stream_core_module.html) module.
+
+Please refer to your OS documentation to learn how to configure nginx, and where its config files are located.
+
+## http server: websocket proxying
+
+First, configure Soju to listen for websocket on the `/run/soju/ws.sock` unix domain socket:
+```
+…
+listen ws+unix:///run/soju/ws.sock
```

Comment:
In general `http+unix://` should be preferred so that file upload works fine as well. (This may require additional `location` blocks in the nginx config?)

Reply:
good point, I haven't implemented file uploads on my instance, and it didn't pass my mind. let me see what's needed.

Reply:
just to make it clear, `listen http+unix:///run/soju/ws.sock` will handle both `/socket` AND `/uploads` right?

Reply:
Yup!